This notebook uses BeautifulSoup to read the postal codes, boroughs, and neighborhoods of Toronto from the Wikipedia page and stores them in a Pandas dataframe.

In [1]:
import pandas as pd
import urllib.request
from bs4 import BeautifulSoup

In [2]:
url = 'https://en.wikipedia.org/wiki/List_of_postal_codes_of_Canada:_M'
page = urllib.request.urlopen(url)
soup = BeautifulSoup(page, "lxml")

postal_table=soup.find('table', class_='wikitable sortable')

In [3]:
postalcode=[]
borough=[]
neighborhood=[]

for row in postal_table.findAll('tr'):
    cells=row.findAll('td')
    if len(cells)==3:
        
# ignore all rows where the borough is 'Not assigned'
        if cells[1].find(text=True).strip('\n') != 'Not assigned':
            postalcode.append(cells[0].find(text=True).strip('\n'))
            borough.append(cells[1].find(text=True).strip('\n'))
            
# if the neighborhood is 'Not assigned' set it to the name of its borough
            if cells[2].find(text=True).strip('\n') == 'Not assigned':
                neighborhood.append(cells[1].find(text=True).strip('\n'))
            else:
                neighborhood.append(cells[2].find(text=True).strip('\n'))

In [4]:
df=pd.DataFrame(columns=['PostalCode', 'Borough', 'Neighborhood'])
df['PostalCode'] = postalcode
df['Borough'] = borough
df['Neighborhood'] = neighborhood

In [5]:
df.head(11)

PostalCode           Borough                                 Neighborhood
0         M3A        North York                                    Parkwoods
1         M4A        North York                             Victoria Village
2         M5A  Downtown Toronto                    Regent Park, Harbourfront
3         M6A        North York             Lawrence Manor, Lawrence Heights
4         M7A  Downtown Toronto  Queen's Park, Ontario Provincial Government
5         M9A         Etobicoke      Islington Avenue, Humber Valley Village
6         M1B       Scarborough                               Malvern, Rouge
7         M3B        North York                                    Don Mills
8         M4B         East York              Parkview Hill, Woodbine Gardens
9         M5B  Downtown Toronto                     Garden District, Ryerson
10        M6B        North York                                    Glencairn

In [6]:
df.shape

(103, 3)

After removing the postal codes with no borough assigned, there are 103 postal codes remaining.